post-chaps changelog version bumps

JaredCorduan opened the pull request:
# Description

This is a post-chaps version bump of the changelogs.

Additionally, I've created testsuite changelogs, which are only to be used to indicate if a release is needed. See the note/change I made in the `RELEASING.md`.

I will keep this PR as draft until the [CHaPs releases](https://github.com/input-output-hk/cardano-haskell-packages/pull/319) are finalized.

TODO - add cardano-ledger-binary #3470

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

lehins commented on libs/cardano-ledger-test/CHANGELOG.md:
```diff
@@ -1 +1,5 @@
+
+## 1.1.0.1
+
+*
```
This one really doesn't need a changelog, since it is neither versioned nor released
```suggestion
## This package is not versioned, so there's no need to update this changelog
```

JaredCorduan replied:
did we deprecate the package? I do see it in CHaPs: https://github.com/input-output-hk/cardano-haskell-packages/tree/main/_sources/cardano-ledger-test

lehins replied:
No, we didn't deprecated it. This was just wrongfully released initially. It is only needed for testing, no one depends on it. So, there is no point of going through the trouble of versioning and releasing it.

JaredCorduan replied:
great, thanks for clearing that up for me!

lehins reviewed (approved):
One minor change is need, otherwise looks good.

lehins commented on libs/cardano-protocol-tpraos/CHANGELOG.md:
```diff
@@ -1,9 +1,22 @@
+
+* upper bounds on cardano-ledger-core and cardano-ledger-shelley
+
+## 1.0.3.1
```
There is also no `1.0.3.1` release, but there is `1.0.3.0`, which is not in the changelog:
```suggestion
## 1.0.3.0
```

JaredCorduan replied:
`1.0.3.1` is what I am trying to release now: https://github.com/input-output-hk/cardano-haskell-packages/pull/319. so `1.0.3.2` is next smallest next bump.

This changelog seems to have been neglected, so I dug through the git history, but maybe I made a mistake. I think the real problem is that I listed `1.0.3.1` twice. the "lower bound" entry should be under `1.0.3.0`.

lehins replied:
Yes, I was looking only at versions. I don't remember the actual changes. `1.0.3.1` was listed twice, so I suggested to change one of them.

lehins commented on libs/cardano-protocol-tpraos/CHANGELOG.md:
```diff
@@ -1,9 +1,22 @@
+
+* lower bound on cardano-crypto-class
+
+## 1.0.1.1
```
There is no release with this version on CHaP, did you mean:
```suggestion
## 1.0.2.0
```

JaredCorduan replied:
hmm, that or `1.0.1.0`...

lehins reviewed (approved):
Noticed a couple of discrepancies.

Commit:
post-chaps changelog version bumps

plus the creation of testsuite changelogs (only to be used to
indicate if a release is needed).
